supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/JesCacheHitCopyingActor.scala

```diff
@@ -29,62 +33,86 @@ case class JesCacheHitCopyingActor(override val jobDescriptor: BackendJobDescrip
   override def copyCachedOutputs(seqOfSimpletons: Seq[WdlValueSimpleton], jobDetritusFiles: Map[String,String],
                                  returnCode: Option[Int]): Future[BackendJobExecutionResponse] = {
 
-      val gcsFileSystem = initializationData.workflowPaths.gcsFileSystemWithUserAuth
-      val jesCallPaths = initializationData.workflowPaths.toJesCallPaths(jobDescriptor.key)
-      val destinationCallRootPath: Path = jesCallPaths.callRootPath
-      val getSourceCallRootPath = jobDetritusFiles.get(jesCallPaths.callRootPathKey) match {
-        case Some(srcPath) => gcsFileSystem.getPath(srcPath)
-        case None => val error = new RuntimeException(s"The call detritus files for source cache hit aren't found for call ${jobDescriptor.call.fullyQualifiedName}")
-      }
+    val gcsFileSystem = initializationData.workflowPaths.gcsFileSystemWithUserAuth
+    val jesCallPaths = initializationData.workflowPaths.toJesCallPaths(jobDescriptor.key)
+    val destinationCallRootPath: Path = jesCallPaths.callRootPath
+    val getSourceCallRootPath = jobDetritusFiles.get(jesCallPaths.CallRootPathKey) match {
+      case Some(srcPath) => gcsFileSystem.getPath(srcPath)
+      case None => new RuntimeException(s"The call detritus files for source cache hit aren't found for call ${jobDescriptor.call.fullyQualifiedName}")
+    }
 
-    def cacheIfPossible = {
+    def cacheIfPossible: Future[BackendJobExecutionResponse] = {
       getSourceCallRootPath match {
         case Some(path: Path) =>
-          copyFiles(path)
+          copyAndRenameFiles(path)
           updateMetadata
           Future(SucceededResponse(jobDescriptor.key, returnCode, createJobOutputs))
-        case error: RuntimeException => Future(FailedNonRetryableResponse(jobDescriptor.key, error, Option(-1)))
+        case error: Throwable => Future(FailedNonRetryableResponse(jobDescriptor.key, error, None))
       }
     }
 
-      def copyFiles(sourceCallRootPath: Path) = {
-        seqOfSimpletons ++ jobDetritusFiles foreach {
-          case WdlValueSimpleton(key, wdlFile: WdlFile) =>
-            copyCachedCallOutputFiles(wdlFile.value, sourceCallRootPath)
-          case (fileName: String, filePath: String) if fileName != jesCallPaths.callRootPathKey =>
-            copyCachedCallOutputFiles(filePath, sourceCallRootPath)
-          case _ =>
-            val error = new RuntimeException(s"Unable to copy cached job outputs for ${jobDescriptor.call.fullyQualifiedName}")
-            FailedNonRetryableResponse(jobDescriptor.key, error, Option(-1))
-        }
+    def copyAndRenameFiles(sourceCallRootPath: Path) = {
+      seqOfSimpletons ++ jobDetritusFiles foreach {
+        case WdlValueSimpleton(key, wdlFile: WdlFile) =>
+          copyFile(wdlFile.value, sourceCallRootPath)
+        case (fileName: String, filePath: String) if fileName != jesCallPaths.CallRootPathKey =>
+          copyFile(filePath, sourceCallRootPath)
```

Comment:
could this be in a `case _ => ...` in the `match` below ?
This way you wouldn't copy things twice and avoid the risk of an pattern match error.

```scala
fileName match {
  case "stdout" => File(gcsFileSystem.getPath(filePath)).copyTo(jesCallPaths.stdoutPath)
  case "stderr" => File(gcsFileSystem.getPath(filePath)).copyTo(jesCallPaths.stderrPath)
  case "returnCode" => File(gcsFileSystem.getPath(filePath)).copyTo(jesCallPaths.returnCodePath)
  case "jesLog" => File(gcsFileSystem.getPath(filePath)).copyTo(jesCallPaths.jesLogPath)
  case _ => copyFile(filePath, sourceCallRootPath)
}
```